addons/account/models/account_move_line.py

```diff
@@ -669,14 +669,23 @@ def _compute_amount_currency(self):
 
     @api.depends('full_reconcile_id.name', 'matched_debit_ids', 'matched_credit_ids')
     def _compute_matching_number(self):
+        large_partials = False
         for record in self:
             if record.full_reconcile_id:
                 record.matching_number = record.full_reconcile_id.name
             elif record.matched_debit_ids or record.matched_credit_ids:
-                record.matching_number = 'P'
+                partials = record.matched_debit_ids or record.matched_credit_ids
+                large_partials = len(partials) > 1
```

Comment:
large_partials might get reassigned from True to False.
protect with condition like
```
if not large_partials: